### PR TITLE
Refactor snackbar to functional widget, add to examples

### PR DESCRIPTION
**Pull request by nicknisi**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

As part of the work for creating a material theme for the snackbar widget dojo/themes#82, I refactored it to be a functional widget and added all of the examples to the new examples app.

### CODE REVIEW

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-qmjpggec5.now.sh
* Tests Deployment: https://dojowidgets-qtfcbhg41.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-qmjpggec5.now.sh
* Tests Deployment: https://dojowidgets-56n1zgtim.now.sh

File: src/snackbar/index.tsx [hunk 1]
```diff
@@ -19,40 +18,34 @@ export interface SnackbarProperties {
 	stacked?: boolean;
 }
 
-@theme(css)
-export class Snackbar extends ThemedMixin(WidgetBase)<SnackbarProperties> {
-	protected render(): DNode {
-		const { type, open, leading, stacked, messageRenderer, actionsRenderer } = this.properties;
+const factory = create({ coreTheme: theme }).properties<SnackbarProperties>();
```
Comment (tomdye) [inline src/snackbar/index.tsx]: why is this `coreTheme` not `theme`? seems odd

Comment (nicknisi) [reply src/snackbar/index.tsx]: Ah, this was just a side-effect of the way I refactored it. I went ahead and updated it!

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-oljbs6owy.now.sh
* Tests Deployment: https://dojowidgets-1cqwgp1xv.now.sh

File: src/snackbar/index.tsx [hunk 2]
```diff
@@ -1,7 +1,6 @@
-import { WidgetBase } from '@dojo/framework/core/WidgetBase';
-import { DNode, RenderResult } from '@dojo/framework/core/interfaces';
-import { theme, ThemedMixin } from '@dojo/framework/core/mixins/Themed';
-import { tsx } from '@dojo/framework/core/vdom';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import theme from '@dojo/framework/core/middleware/theme';
 import * as css from '../theme/snackbar.m.css';
 
 export interface SnackbarProperties {
```
Comment (JamesLMilner) [inline src/snackbar/index.tsx]: @agubler is it good for us to extend `ThemedProperties` for handling custom elements?

Comment (nicknisi) [reply src/snackbar/index.tsx]: You're probably right. I extended `ThemedProperties`.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-n6mkvvcxr.now.sh
* Tests Deployment: https://dojowidgets-43ebun1ht.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-moglifuti.now.sh
* Tests Deployment: https://dojowidgets-76jx9o4mq.now.sh

**Pull request merged**